lib/reader/src/lexer.rs

```diff
@@ -326,6 +338,39 @@ impl<'a> Lexer<'a> {
         }
     }
 
+    fn scan_name(&mut self) -> Result<LocatedToken<'a>, LocatedError> {
+        let loc = self.loc();
+        let begin = self.pos;
+
+        assert!(self.lookahead == Some('%'));
+
+        while let Some(c) = self.next_ch() {
+            if !c.is_alphanumeric() && c != '_' {
```

Comment:
I think in this specific case, we should restrict to ASCII characters only (i.e., reject other Unicode chars). The `FuncName` printer will do the same.

It's not that important if Cretonne accepts Unicode names or not. The `.cton` files are just for writing tests.